crates/conjure_core/src/rule_engine/rule.rs

```diff
@@ -84,43 +83,24 @@ impl Reduction {
     pub fn with_symbols(new_expression: Expression, symbols: SymbolTable) -> Self {
         Self {
             new_expression,
-            new_top: Expression::And(Metadata::new(), Vec::new()),
+            new_top: vec![],
```

Comment:
sometimes we use vec![], sometimes we use ::new - probably better to be consistent.